boards/shared/Src/App/App_SharedSignal.c

```diff
@@ -43,32 +64,75 @@ void App_SharedSignal_Destroy(struct Signal *signal)
     free(signal);
 }
 
-uint32_t App_SharedSignal_GetLastTimeLowMs(const struct Signal *signal)
+uint32_t App_SharedSignal_GetEntryLastTimeLowMs(const struct Signal *signal)
+{
+    return signal->entry_last_time_low_ms;
+}
+
+uint32_t App_SharedSignal_GetEntryLastTimeHighMs(const struct Signal *signal)
+{
+    return signal->entry_last_time_high_ms;
+}
+
+uint32_t App_SharedSignal_GetExitLastTimeLowMs(const struct Signal *signal)
 {
-    return signal->last_time_low_ms;
+    return signal->exit_last_time_low_ms;
 }
 
-uint32_t App_SharedSignal_GetLastTimeHighMs(const struct Signal *signal)
+uint32_t App_SharedSignal_GetExitLastTimeHighMs(const struct Signal *signal)
 {
-    return signal->last_time_high_ms;
+    return signal->exit_last_time_high_ms;
+}
+
+bool App_SharedSignal_IsCallbackTriggered(const struct Signal *const signal)
+{
+    return signal->is_callback_triggered;
 }
 
 void App_SharedSignal_Update(struct Signal *signal, uint32_t current_time_ms)
 {
-    if (signal->is_high(signal->world))
+    if (signal->is_entry_high(signal->world))
     {
-        signal->last_time_high_ms = current_time_ms;
+        signal->entry_last_time_high_ms = current_time_ms;
     }
     else
     {
-        signal->last_time_low_ms = current_time_ms;
+        signal->entry_last_time_low_ms = current_time_ms;
     }
 
-    const uint32_t time_since_last_low =
-        current_time_ms - signal->last_time_low_ms;
+    if (signal->is_exit_high(signal->world))
+    {
+        signal->exit_last_time_high_ms = current_time_ms;
+    }
+    else
+    {
+        signal->exit_last_time_low_ms = current_time_ms;
+    }
 
-    if (time_since_last_low >= signal->callback.high_duration_ms)
+    if (!signal->is_callback_triggered)
     {
-        signal->callback.function(signal->world);
+        const uint32_t entry_time_since_last_low =
+            current_time_ms - signal->entry_last_time_low_ms;
```

Comment:
Should this be called something like _entry_signal_high_count_ or _entry_signal_high_duration_ and possibly rename _entry_high_duration_ms_ to _entry_high_duration_threshold_ms_ , or something along those lines?

Reply:
Hmm yeah reading over this again it doesn't seem very clear what i was trying to describe. I changed it to `time_since_is_entry_last_low` and `time_since_is_exit_last_low`